src/runtime_src/core/include/xgq.h

```diff
@@ -124,7 +134,35 @@ struct xrt_com_queue_entry {
 };
 
 #define XGQ_SUB_HEADER_SIZE	(sizeof(struct xrt_sub_queue_entry) - 4)
-#define	XRT_COM_Q1_SLOT_SIZE	(sizeof(struct xrt_com_queue_entry))
+#define XRT_COM_Q1_SLOT_SIZE	(sizeof(struct xrt_com_queue_entry))
+
+/**
+ * struct xrt_cmd_load_xclbin: load XCLBIN command
+ *
+ * @address:	XCLBIN address
+ * @size:	XCLBIN size in Byte
+ * @addr_type:	Address tyep
+ *
+ * This command is used to load XCLBIN to device through XGQ.
+ * This is an indirect command that XCLBIN blob's address is
+ * embedded.
+ */
+struct xrt_cmd_load_xclbin {
+	union {
+		struct {
+			uint32_t opcode:16; /* [15-0]   */
+			uint32_t count:15;  /* [30-16] */
+			uint32_t state:1;   /* [31] */
+			uint16_t cid;
+			uint16_t rsvd;
+		};
+		uint32_t header[2];
```

Comment:
warning: do not declare C-style arrays, use std::array<> instead [hicpp-avoid-c-arrays]

```cpp

```

Reply:
ditto